Fix extra gap and outer scrollbar in subscribers list on channel sett…

JANVI-CHATURVEDI opened the pull request:
This PR fixes the extra bottom gap below nested scrollbar in the channel settings subscribers list, and aligns the layout with the group settings modal’s members section for visual consistency.

The issue was caused by an unnecessary top margin on the loading state that introduced an extra gap at bottom .

**Fixes:** Extra gap below inner scrollbar in subscribers list (reported in the issues section on Zulip chat: [link](https://chat.zulip.org/#narrow/channel/9-issues/topic/Extra.20gap.20after.20subscribers.20list/with/2333797))

**Before**
<img width="1896" height="922" alt="Screenshot (73)" src="https://github.com/user-attachments/assets/45e98d55-aed4-47c8-9e48-91c14bd8f312" />



 - Extra bottom gap visible
 - Nested scrollbar inside the modal


**After**
<img width="1920" height="926" alt="Screenshot (100)" src="https://github.com/user-attachments/assets/664019d2-9792-4a57-9b15-6b730190b64d" />





- Gap removed
- Layout matches other group settings 


## Self-review checklist

* [x] Self-reviewed the changes for clarity and maintainability (variable names, code reuse, readability, etc.).
* [x] Communicated decisions, questions, and potential concerns (e.g., JS height behavior for bottom space).
* [x] Individual commits are ready for review:
  * Commit history follows Zulip commit guidelines.
  * Commit message explains reasoning and motivation for changes.
* [x] Completed manual review and testing of the following:
  * Visual appearance of the changes (including before/after screenshots).
  * Responsiveness and internationalization (if applicable).
  * Strings and tooltips (not applicable for this UI change).
  * End-to-end functionality of buttons, interactions and flows.
  * Corner cases, error conditions, and easily imagined bugs.

JANVI-CHATURVEDI commented:
Hi @sahil839, I’ve updated the commit messages as per Zulip guidelines. Please have a look.

alya commented:
@JANVI-CHATURVEDI How many characters is your commit title, and how many do the guidelines request?

As a heads up, we will be able to review your work only if you *carefully* address all reviewer feedback.

JANVI-CHATURVEDI commented:
Hi @alya, I noticed the previous commit title exceeded the 72-character limit. I’ve shortened it according to the guidelines and updated the PR for review. Thank you!

JANVI-CHATURVEDI commented:
Thanks for the feedback, @alya!
I’ve now updated the PR description with new before/after screenshots taken at the same window size and scroll position.  
Happy to make any further changes if needed.

JANVI-CHATURVEDI commented:
Hi @alya, while testing with longer subscriber lists in other settings, I realized the reported issue was about the extra gap and the unnecessary outer scrollbar. I’ve adjusted the fix accordingly and updated the PR description, title, and screenshots to reflect this. I’m happy to make any further changes if needed. Thanks!

alya commented:
Why is there still extra blank space on the bottom? Can it be reduced?

JANVI-CHATURVEDI commented:
hi @alya , I looked into this further and found that the remaining space isn’t coming from CSS. It’s due to the list height being set via a JS-calculated max-height, which can result in small variations depending on browser layout timing, even on the same screen.

I’ve reduced the unnecessary spacing, but fully removing this would require changing the height calculation logic itself, which could risk regressions in other views.

alya commented:
Please update the PR description with the latest screenshots for review.

JANVI-CHATURVEDI commented:
Hi @alya , I’ve updated the PR description with the latest screenshots.
I also wanted to clarify one thing about the bottom spacing: the height of the subscriber list is calculated in JS, so depending on when that calculation runs and the browser layout state, the modal can sometimes render with a small bottom gap and sometimes without it. This PR doesn’t change that logic.

timabbott commented:
Merged, thanks for the contribution @JANVI-CHATURVEDI and review @sahil839 !

@JANVI-CHATURVEDI I hope you learned a lot about fixing bugs cleanly :)
